### PR TITLE
[BH-1243] Fix arrows behaviour in spinners

### PR DESCRIPTION
**Pull request by tigransoghbatyan-digica**
Refactor and fix arrows visibility in spinners

### CODE REVIEW

File: products/BellHybrid/apps/application-bell-settings/models/alarm_settings/SnoozeListItemProvider.cpp [hunk 1]
```diff
@@ -83,16 +75,11 @@ namespace app::bell_settings
...
                 chimeRange.push_back(NumWithStringListItem::Value{utils::translate("app_alarm_clock_no_snooze")});
-                for (unsigned int i = 1; i <= val; i++) {
-                    if ((i != Intervals::Interval_1) && (i != Intervals::Interval_2) && (i != Intervals::Interval_5) &&
-                        (i != Intervals::Interval_10) && (i != Intervals::Interval_15) &&
-                        (i != Intervals::Interval_30)) {
-                        continue;
-                    }
-                    chimeRange.push_back(NumWithStringListItem::Value{i, minStr});
-                }
+                std::for_each(allowedIntervals.begin(), allowedIntervals.lower_bound(val), [&](auto val) {
```
Comment (tomaszrybarski-digica) [inline products/BellHybrid/apps/application-bell-settings/models/alarm_settings/SnoozeListItemProvider.cpp:78]: What happens when, for example: val == 12 (snooze length set in previous window), won't std::lower_bound point to next greater than 12 in the set, which is 15 and add it to the allowable range for chime interval? i think it is not supposed to work this way, for a val of 12 the allowable max. limit for interval should be 10 (always lower than chime length)?

**Review by swichu91 — COMMENTED** [review]
I'm afraid it does not solve the problem entirely. Please see the discussion here https://appnroll.slack.com/archives/CCMBKV2NQ/p1637666690009300 .

File: products/BellHybrid/apps/application-bell-settings/models/alarm_settings/SnoozeListItemProvider.cpp (hunk 1 above)
Comment (tigransoghbatyan-digica) [reply products/BellHybrid/apps/application-bell-settings/models/alarm_settings/SnoozeListItemProvider.cpp:78]: No, the lower bound is choosing a lower value between the two closest in the set

Comment (tomaszrybarski-digica) [reply products/BellHybrid/apps/application-bell-settings/models/alarm_settings/SnoozeListItemProvider.cpp:78]: In that case it's fine, i was looking at https://www.geeksforgeeks.org/set-lower_bound-function-in-c-stl/ - maybe there is an error there

Comment (tigransoghbatyan-digica) [reply products/BellHybrid/apps/application-bell-settings/models/alarm_settings/SnoozeListItemProvider.cpp:78]: can we resolve this conversation?

**Comment by tigransoghbatyan-digica** [comment]
> I'm afraid it does not solve the problem entirely. Please see the discussion here https://appnroll.slack.com/archives/CCMBKV2NQ/p1637666690009300 .

I have no access to this conversation

**Review by mjan-dev — CHANGES_REQUESTED** [review]
Blocking until new branch for patch 0 will be created. It is not of highest priority.

**Review by mjan-dev — APPROVED** [review]
As @swichu91 mentioned, it does not solve the problem, so the task cannot be set to done. However, it should improve the behavior nevertheless. I see no harm in merging it.

**Comment by mjan-dev** [comment]
This behavior was already fixed in other PRs. Closing.

**Pull request closed without merging**